lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -636,24 +650,21 @@ def do_3d_projection(self, renderer):
 
         # if required sort by depth (furthest drawn first)
         if self._zsort:
-            z_segments_2d = sorted(
-                ((self._zsortfunc(zs), np.column_stack([xs, ys]), fc, ec, idx)
-                 for idx, ((xs, ys, zs), fc, ec)
-                 in enumerate(zip(xyzlist, cface, cedge))),
-                key=lambda x: x[0], reverse=True)
+            z_argsort = np.argsort(
+                [self._zsortfunc(xyz[:, 2]) for xyz in xyzlist])[::-1]
```

Comment:
We should double-check if argsorting is the right approach here for this. numpy uses a different algorithm for argsorting than it does for regular sorting, IIRC. Also, argsorting suffers from non-locality of data as the sorting process wraps up, which causes cache-misses.

Reply:
According to the docs, `np.argsort` and `np.sort` use quicksort while the builtin `sorted` uses timsort. Timsort [is very fast for pre-sorted data](https://stackoverflow.com/a/19587279/1791279) while there have been report of quicksort being faster implementation wise for random data https://github.com/numba/numba/issues/1383#issuecomment-134539513 .

Is there reason to believe that data would be pre-sorted here? or that this is a performance critical part?